### PR TITLE
LogFactory: Add generic-type versions of GetLogger() and GetCurrentClassLogger()

### PR DESCRIPTION
**Pull request by bhaeussermann**
Related to #1203

### CODE REVIEW

**Comment by 304NotModified** [comment]
I restarted appveyor build 3 times and same unit test fails. But dunno why as it looks unrelated.

**Comment by 304NotModified** [comment]
Please merge (rebase would be even better)  with master.

File: src/NLog/LogFactory.cs [hunk 1]
```diff
@@ -351,6 +351,24 @@ public Logger GetCurrentClassLogger()
         }
 
         /// <summary>
+        /// Gets the logger with the name of the current class. 
+        /// </summary>
+        /// <returns>The logger.</returns>
+        /// <remarks>This is a slow-running method. 
+        /// Make sure you're not doing this in a loop.</remarks>
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public T GetCurrentClassLogger<T>() where T : Logger
```
Comment (304NotModified) [inline src/NLog/LogFactory.cs]: should it not be `T: ILogger`?

Comment (bhaeussermann) [inline src/NLog/LogFactory.cs]: Good call!

I just tried it, but it yields the compiler error `Constraint type 'ILogger' is not CLS-complaint`

Apparently only CLS-compliant types can be used for generic types, and `ILogger` is marked with `[CLSCompliant(false)]`.

Comment (bhaeussermann) [inline src/NLog/LogFactory.cs]: Besides, all of the other `Get...Logger()` methods return type `Logger`.

Comment (304NotModified) [inline src/NLog/LogFactory.cs]: CLS compliant is not needed for generic types, but it is for this class (as it's not marked als CLScompliant false). 

Thanks for checking! 

PS. The return type is here `T` ;)

**Comment by 304NotModified** [comment]
Thanks again!

**Pull request merged**